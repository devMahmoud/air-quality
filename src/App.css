body {
  margin: 0;
  padding: 0;
  background-color: #eb4c8b;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.link {
  text-decoration: none;
  color: #fff;
}

.cities-list {
  color: #fff;
}

.country-container {
padding: 1rem;
}

.country-container:nth-child(even) {
  background-color: #ce4277;
}

.country-content {
  display: flex;
  justify-content: space-between;
}

.country-name {
  font-size: larger;
  font-weight: 600;
}
