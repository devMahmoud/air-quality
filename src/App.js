import { Routes, Route } from 'react-router-dom';
import CountriesList from './components/CountriesList';
import './App.css';
import Navbar from './components/NavBar';
import CountryDetails from './components/CountryDetails';

function App() {
  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={<CountriesList />} />
        <Route path="/Country/:countryName" element={<CountryDetails />} />
      </Routes>
    </div>
  );
}

export default App;
