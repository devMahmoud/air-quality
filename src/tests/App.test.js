import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import testStore from './TestStore';
import App from '../App';

const store = testStore();

describe('Test the App components', () => {
  it('renders the App correctly', () => {
    const app = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });

  it('shows the CountryDetails page for specific element when clicking the element link', async () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
    );
    fireEvent.click(screen.getByText('Albania'));
    expect(container.getElementsByClassName('details-wrapper')).toBeTruthy();
  });
});
