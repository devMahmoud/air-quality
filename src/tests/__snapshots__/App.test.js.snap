// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App components renders the App correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="home"
        >
          <nav
            class="nav-container"
          >
            <button
              class="back-btn btn"
              type="button"
            >
              ❰
            </button>
            <select
              id="views"
              name="views"
            >
              <option
                value="confirmed"
              >
                Confirmed Cases View
              </option>
              <option
                value="deaths"
              >
                Deaths View
              </option>
            </select>
            <div
              class="icons"
            >
              <img
                alt=""
                class="mic-icon"
                src="mic-20.png"
              />
              <button
                class="setting-btn btn"
                type="button"
              >
                ⚙
              </button>
            </div>
          </nav>
          <div
            class="cities-list"
          >
            <div
              class="country-wrapper"
            >
              <a
                class="link"
                href="/Country/Albania"
              >
                <div
                  class="country-container"
                >
                  <div
                    class="country-content"
                  >
                    <div
                      class="country-name"
                    >
                      Albania
                    </div>
                    <div
                      class="status"
                    >
                      <p
                        class="confirmed"
                      >
                        Confirmed Cases
                        <br />
                        277940
                      </p>
                    </div>
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="home"
      >
        <nav
          class="nav-container"
        >
          <button
            class="back-btn btn"
            type="button"
          >
            ❰
          </button>
          <select
            id="views"
            name="views"
          >
            <option
              value="confirmed"
            >
              Confirmed Cases View
            </option>
            <option
              value="deaths"
            >
              Deaths View
            </option>
          </select>
          <div
            class="icons"
          >
            <img
              alt=""
              class="mic-icon"
              src="mic-20.png"
            />
            <button
              class="setting-btn btn"
              type="button"
            >
              ⚙
            </button>
          </div>
        </nav>
        <div
          class="cities-list"
        >
          <div
            class="country-wrapper"
          >
            <a
              class="link"
              href="/Country/Albania"
            >
              <div
                class="country-container"
              >
                <div
                  class="country-content"
                >
                  <div
                    class="country-name"
                  >
                    Albania
                  </div>
                  <div
                    class="status"
                  >
                    <p
                      class="confirmed"
                    >
                      Confirmed Cases
                      <br />
                      277940
                    </p>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
